src/main/java/com/unmoon/ToolRequiredPlugin.java

```diff
@@ -178,6 +183,27 @@ else if (entry.getOption().startsWith("Cut") && (target.contains("ree") || cutOv
 			if (config.mine() && entry.getOption().equals("Mine") && !ANY_PICKAXE.fulfilledBy(playerItems)) {
 				root.removeMenuEntry(entry);
 			}
+            // Check if farming is enabled, and check if player has magic secateurs
+            if (config.farm() && !MAGIC_SECATEURS.fulfilledBy(playerItems)) {
+                String target = removeTags(entry.getTarget());
+                // entry option starts with Pick OR Harvest AND contains one of the items affected by the secateurs
+                if ((entry.getOption().startsWith("Pick") || entry.getOption().startsWith("Harvest"))
+                        && farmingOverrides.stream().anyMatch(target::contains)) {
+                    root.removeMenuEntry(entry);
+                }
+            }
 		}
 	}
+
+    @Subscribe
+    public void onGameStateChanged(GameStateChanged event) {
+        // listen for game state changes
+        GameState state = event.getGameState();
+
+        // detect login event AND check loginMessage is True - send login message with version number
+        if (state == GameState.LOGGED_IN && config.loginMessage()) {
+            String message = "Tool Required plugin v" + config.pluginVersion();
```

Comment:
The intent of this message is to announce new changes in the plugin, i.e. when plugin is updated to `1.2.0`, it should say something like `Tool Required updated: Farming patches added (magic secateurs required). Click here to hide this message in the future.`